tgui/packages/tgui/interfaces/Fabrication/AnimatedQuantityLabel.tsx

```diff
@@ -0,0 +1,115 @@
+import { Component, createRef } from 'inferno';
+import { formatSiUnit } from '../../format';
+
+/**
+ * The properties of an animated quantity label.
+ */
+export interface AnimatedQuantityLabelProps {
+  /**
+   * The target value to approach.
+   */
+  targetValue: number;
+}
+
+/**
+ * Quantity labels are animated at roughly 60 frames per second.
+ */
+const SIXTY_HZ = 1_000.0 / 60.0;
+
+/**
+ * An animated quantity label. Shows an SI-encoded number, and animates it
+ * towards towards the provided target value.
+ */
+export class AnimatedQuantityLabel extends Component<AnimatedQuantityLabelProps> {
+  /**
+   * The inner `<span/>` being updated sixty times per second.
+   */
+  protected ref = createRef<HTMLSpanElement>();
```

Comment:
Accessor modifiers are noise, because component classes are never meant to be extended, and its methods can never be accessed from outside. Inheritance is bad.

You should not have them here at all.